previewers/betatest/js/video.js

```diff
@@ -9,5 +9,104 @@ function translateBaseHtmlPage() {
 
 function writeContent(fileUrl, file, title, authors) {
     addStandardPreviewHeader(file, title, authors);
-    $(".preview").append($("<video/>").prop("controls",true).append($('<source/>').attr("src",fileUrl)));
+
+    const queryParams = new URLSearchParams(window.location.search.substring(1));
+    const id = queryParams.get("datasetid");
+    const siteUrl = queryParams.get("siteUrl");
+    const versionUrl = `${siteUrl}/api/datasets/${id}/versions/`
+        + queryParams.get("datasetversion");
+    const videoId =queryParams.get("fileid") * 1; // converted to number
+    const userLanguages = [...navigator.languages];
+    const locale = queryParams.get("locale");
+    if (locale && !userLanguages.includes(locale)) {
+        userLanguages.unshift(locale); // add as first element
+    }
+
+    $.ajax({
+        type: 'GET',
+        dataType: 'json',
+        crosssite: true,
+        url: versionUrl,
+        success: function(data, status) {
+            appendVideoElements(fileUrl, videoId, data.data.files, siteUrl, userLanguages);
+        },
+        error: function(request, status, error) {
+            // fallback to simple video element
+            $(".preview").append($("<video/>") .prop("controls", true) .append($('<source/>').attr("src", fileUrl)))
+        }
+    });
+}
+
+function appendVideoElements(fileUrl, videoId, files, siteUrl, userLanguages) {
+
+    const baseName = files // the video file name without extension
+        .find(item => item.dataFile.id === videoId)
+        .label.replace(/\.[a-z0-9]+$/i,'');
+
+    // find labels like "baseName.en.vtt", "baseName.de-CH.vtt" or "baseName.vtt"
+    const regex = new RegExp(`${baseName}(\\.([-a-z]+))?\\.vtt$`, 'i')
+
+    // create a map of URLs with their (optional) language
+    const subtitles = files
+        .filter(item => regex.test(item.label))
+        .reduce((map, item) => {
+            const lang = item.label.match(regex)[2];
+            const url = `${siteUrl}/api/access/datafile/${item.dataFile.id}?gbrecs=true&amp;key=93423e09-848c-47cb-a979-219dafcfa4da`;
```

Comment:
hardcoded key? It looks like this will have to run with the apikey sent in (for draft versions anyway) rather than signedURLs because we can't currently support allowing you to get signedUrls for other files? If so, it might be better to make this a separate previewer (videoWithCaptions.js?) or maybe just use the presence of the API key ( just for non-draft datasets/non-restricted files) as a way to decide whether to try and get the captions?

Reply:
Dropping the API key resulted in desired behavior. When logged in, a restricted video appears with subtitles. When not logged in and viewing a not restricted video with restricted subtitles, the menu shows available languages but the subtitles don't appear.